src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java

```diff
@@ -27,20 +29,36 @@
 public class FinancialTransactionServiceImpl implements FinancialTransactionService {
 
     private final FinancialTransactionRepository financialTransactionRepository;
+
     private final FinancialTransactionModelMapper financialTransactionModelMapper;
+
     private final WalletRepository walletRepository;
+
+    private final FinancialTransactionCategoryRepository financialTransactionCategoryRepository;
+
     @Override
     public FinancialTransactionDTO createFinancialTransaction(@Valid FinancialTransactionCreateDTO financialTransactionCreateDTO) {
         Long walletId = financialTransactionCreateDTO.walletId();
         Wallet wallet = walletRepository.findById(walletId).orElseThrow(() -> {
             throw new AppRuntimeException(ErrorCode.W003, String.format("Wallet with id: %d does not exist", walletId));
         });
+        FinancialTransactionCategory financialTransactionCategory = null;
+        Long categoryId = financialTransactionCreateDTO.categoryId();
+
+        if (categoryId != null) {
+            financialTransactionCategory = financialTransactionCategoryRepository.findById(categoryId)
+                    .orElseThrow(() -> {throw new AppRuntimeException(ErrorCode.FTC001,
+                        String.format("Financial transaction category with id: %d does not exist", categoryId));
+            });
+        }
```

Comment:
Form 45 to 53 you can extract private method f.e: private FinancialTransactionCategory findFinancialTransactionCategory(FinancialTransactionCreateDTO financialTransactionCreateDTO)